### PR TITLE
Initial version of TableUnspanner

### PR DESCRIPTION
**Pull request by chrismytton**
This is a port of the `<table>` unspanning code from https://github.com/everypolitician-scrapers/turkey-tbmm-wikipedia/pull/3.

# Notes to reviewer

You can see this in action in https://github.com/everypolitician-scrapers/turkey-tbmm-wikipedia/pull/3.

The naming is currently a bit odd, the module is called `TableUnspanner` and that in turn contains a `UnspannedTable` class. I _think_ this makes sense, `TableUnspanner` is the library name, `UnspannedTable` is the object that represents a table that has been unspanned. However I am open to persuasion that this should be changed :)

### CODE REVIEW

File: test/table_unspanner_test.rb [hunk 1]
```diff
@@ -1,11 +1,32 @@
...
+        <tr>
+          <td>Alice</td>
+          <td rowspan="2">Test subject</td>
+        </tr>
+        <tr>
+          <td>Bob</td>
+        </tr>
+      </table>
+      TABLE
+
```
Comment (tmtmtmtm) [inline test/table_unspanner_test.rb:25]: I think you need a test that has at least one colspan, and probably at least one span in a `th` as well.

**Review by tmtmtmtm — CHANGES_REQUESTED** [review]
As discussed on slack, I'm also a little concerned that the interface here is being driven too much by its use as a Decorator, rather than this being a good standalone library in its own right.

**Comment by tmtmtmtm** [comment]
I'm assuming this version can be closed now?

**Comment by chrismytton** [comment]
Ah yep, indeed it can!

**Pull request closed without merging**